main.go

```diff
@@ -22,17 +23,21 @@ import (
 )
 
 var (
-	pass = color.FgGreen
-	skip = color.FgYellow
-	fail = color.FgHiRed
+	pass   = color.FgGreen
+	skip   = color.FgYellow
+	fail   = color.FgHiRed
+	ignore = flag.Bool("skipnotest", false, "skip packages with no test files")
 )
 
 const paletteEnv = "GOTEST_PALETTE"
 
 func main() {
+	flag.Parse()
```

Comment:
Using `flag.Parse` causes unexpected behavior in `go test`. For example, obtaining verbose output from `go test` requires the `-v` flag to be used after non-flag arguments. Specifically,
`gotest ./... -v` works, but `gotest -v ./...` doesn't.

We should use a `flag.FlagSet` with `flag.ContinueOnError` so that we can avoid exiting when we attempt to parse flags intended for `go test`.

Reply:
Unfortunately, it doesn't look like the flag package provides a typed error for unknown flags, so we're stuck with string parsing, which I'm not a huge fan of for error identification.

https://golang.org/src/flag/flag.go?s=31609:31672#L935

Reply:
Even with `ContinueOnError`, flag parsing is halted on the first unknown flag.